scaffoldmaker/meshtypes/meshtype_3d_colonsegmentteniacoli1.py

```diff
@@ -1013,38 +1057,42 @@ def getTeniaColi(region, nodeIdentifier, elementIdentifier, useCrossDerivatives,
 
     # Calculate texture coordinates and derivatives
     nodeIdentifier = prevNodeIdentifier + 1
-    dTC = uList[1] - uList[0]
 
-    for N in range(4):
+    for N in range(tcCount + 1): # 4
         idxTCMid = N*(elementsCountAroundTC + elementsCountAroundHaustrum)
         TCStartIdx = idxTCMid - int(elementsCountAroundTC*0.5)
         TCEndIdx = idxTCMid + int(elementsCountAroundTC*0.5)
+        dTC = uList[idxTCMid] - uList[TCStartIdx] if N > 0 else uList[TCEndIdx] - uList[idxTCMid]
         v1 = [uList[TCStartIdx], 0.0, 1.0] if N > 0 else [-dTC, 0.0, 1.0]
         v2 = [  uList[idxTCMid], 0.0, 2.0]
-        v3 = [  uList[TCEndIdx], 0.0, 1.0] if N < 3 else [ 1 + dTC , 0.0, 1.0]
+        v3 = [  uList[TCEndIdx], 0.0, 1.0] if N < tcCount else [ 1 + dTC , 0.0, 1.0]
         d1 = d2 = d3 = [dTC, 0.0, 0.0]
         nx = [v1, v2, v3]
         nd1 = [d1, d2, d3]
-        sx, _, _, _, _ = interp.sampleCubicHermiteCurves(nx, nd1, elementsCountAroundTC)
```

Comment:
For future reference, it's generally better to change this to:
sx = interp.sampleCubicHermiteCurves(nx, nd1, elementsCountAroundTC)[0]
This allows individual values to be picked out, and is immune to changes in functions returning more values.
Use _ for in-between return values.